src/Security/src/Authorization.Certificate/CertificateAuthorizationBuilderExtensions.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Authentication.Certificate;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Steeltoe.Common;
+using Steeltoe.Common.Certificates;
+using Steeltoe.Common.Configuration;
+
+namespace Steeltoe.Security.Authorization.Certificate;
+
+public static class CertificateAuthorizationBuilderExtensions
+{
+    /// <summary>
+    /// Adds the necessary components and policies for server-side authorization of application instance identity certificates.
+    /// <para>
+    /// Components include <see cref="CertificateOptions" /> named "AppInstanceIdentity" and certificate forwarding.
+    /// </para>
+    /// <para>
+    /// Secure your endpoints with the included authorization policies by referencing <see cref="CertificateAuthorizationPolicies" />.
+    /// </para>
+    /// </summary>
+    /// <param name="authorizationBuilder">
+    /// The <see cref="AuthorizationBuilder" />.
+    /// </param>
+    public static AuthorizationBuilder AddAppInstanceIdentityCertificate(this AuthorizationBuilder authorizationBuilder)
```

Comment:
The way I understand it (please correct me if I got it wrong), this is the purpose of this extension:

"Defines a policy that verifies the space/org in the incoming client certificate matches the space/org of the local application instance identity certificate in configuration."

How it should be used:
- The caller needs to ensure valid incoming certificate using ASP.NET's `services.AddAuthentication().AddCertificate()`
- The caller needs to ensure certificate paths are in configuration (by calling our `IConfigurationBuilder.AddAppInstanceIdentityCertificate`)
- For convenience, this extension also activates change detection in configuration paths
- The caller needs to annotate endpoints using `[Authorize(Policy = ...)]` to activate the policy

Suggested name: AddSameSpaceSameOrgCertificatePolicy (because built-in names are centered around Add*Policy)

Reply:
Your understanding is accurate, how about `AddSpaceAndOrgPolicies`?

Reply:
Ok